src/main/java/io/github/apace100/apoli/power/factory/condition/DistanceFromCoordinatesConditionRegistry.java

```diff
@@ -0,0 +1,197 @@
+package io.github.apace100.apoli.power.factory.condition;
+
+import io.github.apace100.apoli.Apoli;
+import io.github.apace100.apoli.data.ApoliDataTypes;
+import io.github.apace100.apoli.util.Comparison;
+import io.github.apace100.apoli.util.Shape;
+import io.github.apace100.calio.data.SerializableData;
+import io.github.apace100.calio.data.SerializableDataType;
+import io.github.apace100.calio.data.SerializableDataTypes;
+import net.minecraft.block.pattern.CachedBlockPosition;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldView;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+/**
+ * @author Alluysl
+ * Handles the registry of the distance_from_spawn condition in both block and entity conditions to avoid duplicating code.
+ * */
+public class DistanceFromCoordinatesConditionRegistry {
+
+    private static final ArrayList<Object> previousWarnings = new ArrayList<>();
+    private static void warnOnce(String warning, Object key){
+        if (!previousWarnings.contains(key)){
+            previousWarnings.add(key);
+            Apoli.LOGGER.warn(warning);
+        }
+    }
+    private static void warnOnce(String warning){ warnOnce(warning, warning); }
+
+    /**
+     * Warns the user of an issue getting an information needed for expected behavior, but only once (doesn't spam the console).
+     * @param object the object that couldn't be acquired
+     * @param from the object that was supposed to provide the required object
+     * @param assumption the result assumed because of the lack of information
+     * @return the assumed result
+     * */
+    private static <T> T warnCouldNotGetObject(String object, String from, T assumption){
+        warnOnce("Could not retrieve " + object + " from " + from + " for distance_from_spawn condition, assuming " + assumption + " for condition.");
+        return assumption;
+    }
+
+    /**
+     * Returns an array of aliases for the condition.
+     * */
+    private static String[] getAliases(){
+        return new String[]{"distance_from_spawn", "distance_from_coordinates"};
+    }
+
+    private static SerializableData getSerializableData(String alias){
+        // Using doubles and not ints because the player position is a vector of doubles and the sqrt function (for the distance) returns a double so we might as well use that precision
+        return new SerializableData()
+            .addFunctionedDefault("reference", SerializableDataTypes.STRING, data -> alias.equals("distance_from_coordinates") ? "world_origin" : "world_spawn") // the reference point
```

Comment:
No need to use `addFunctionedDefault` here - the `alias` is a String provided to the method and therefore "constant". You only need to check it once and then provide a different default string value to the regular `add` method based on what it is.
`addFunctionedDefault` is for when you want a different default value based on the values of other fields provided by the user.